src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx

```diff
@@ -510,39 +514,35 @@ double BostedChristyEMPXSec::XSec(
     pF = 0.5*kF;
     double F1d;
     FermiSmearingA(Q2, W, pF, Es, F1p, F1d, sigmaT, sigmaL);
+    R = 0.;
     if(sigmaT>0.) 
       R = sigmaL/sigmaT;
     W1 = (2.*Z*F1d + (A - 2.*Z)*(2.*F1d - F1p))/MN; 
 
     W1 *= (fAfitcoef[0] + x*(fAfitcoef[1] + x*(fAfitcoef[2] + x*(fAfitcoef[3] + x*(fAfitcoef[4] + x*fAfitcoef[5])))));
 
     if(W>0.) 
-      W1 *= TMath::Power(fAfitcoef[6] + (fAfitcoef[7]*W + fAfitcoef[8]*W2)/(fAfitcoef[9] + fAfitcoef[10]*Q2),2);
+      W1 *= TMath::Power(fAfitcoef[6] + (fAfitcoef[7]*W + fAfitcoef[8]*Wsq)/(fAfitcoef[9] + fAfitcoef[10]*Q2),2);
       
     double F1M = MEC2009(A, Q2, W);
 
     W1 += F1M;
-    if(W2>0.) 
+    if(Wsq>0.) 
       R *= (fAfitcoef[11] + fAfitcoef[12]*A);
   }
   
   double emcfac = FitEMC(x, A);
 
-  double F1 = MN*W1*emcfac;  
+  W1 *= emcfac;
     
   double nu2 = nu*nu;
-  double K = (W2 - MN2)/2./MN;
   double Eprime = E - nu;
   double sin2theta_2 = Q2/4/E/Eprime;
   double cos2theta_2 = 1 - sin2theta_2;
-  double tan2theta_2 = sin2theta_2/cos2theta_2;
-  double eps = 1./(1. + 2*(1+nu2/Q2)*tan2theta_2);              // Ref.1, Eq. (4)
-  double Gamma = kAem*Eprime*K/Q2/E/(1-eps)/2/kPi2;             // Ref.1, Eq. (5)
-  sigmaT = 4*kPi2*kAem*F1/K/MN;
-  sigmaL = R*sigmaT;
-  double xsec = Gamma*(sigmaT+eps*sigmaL);                      // Ref.1, Eq. (3) d2xsec/dOmegadEprime
+  double W2 = W1*(1 + R)/ (1+nu2/Q2);
```

Comment:
I can't comment on these changes but I'll assume that you've verified them

Reply:
Yes, I have verified them. I just rewrote the formulas in more compact form